presto-main/src/main/java/com/facebook/presto/type/setdigest/SetDigest.java

```diff
@@ -165,22 +165,16 @@ public static long exactIntersectionCardinality(SetDigest a, SetDigest b)
 
     public static double jaccardIndex(SetDigest a, SetDigest b)
     {
-        int sizeOfSmallerSet = Math.min(a.minhash.size(), b.minhash.size());
         LongSortedSet minUnion = new LongRBTreeSet(a.minhash.keySet());
         minUnion.addAll(b.minhash.keySet());
 
         int intersection = 0;
-        int i = 0;
         for (long key : minUnion) {
```

Comment:
This is iterating over all elements in minUnion and then checking for the element's membership in each input set.
How about just iterating over the elements in the smaller set among the two, and then checking if the element exists  in the other?